packages/ra-ui-materialui/src/input/DateInput.tsx

```diff
@@ -86,43 +84,47 @@ export const DateInput = ({
     const hasFocus = React.useRef(false);
 
     // update the input text when the user types in the input
-    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        if (onChange) {
-            onChange(event);
-        }
-        if (
-            typeof event.target === 'undefined' ||
-            typeof event.target.value === 'undefined'
-        ) {
-            return;
-        }
-        const target = event.target;
-
-        const newValue =
-            target.valueAsDate !== undefined &&
-            target.valueAsDate !== null &&
-            !isNaN(new Date(target.valueAsDate).getTime())
-                ? parse
-                    ? parse(target.valueAsDate)
-                    : getStringFromDate(target.valueAsDate)
-                : parse
-                  ? parse(target.value)
-                  : getStringFromDate(target.value);
-
-        // Some browsers will return null for an invalid date so we only change react-hook-form value if it's not null
-        // The input reset is handled in the onBlur event handler
-        if (newValue !== '' && newValue != null) {
-            field.onChange(newValue);
-            valueChangedFromInput.current = true;
+    const handleChange = useEvent(
```

Comment:
I'm surprised you didn't need to perform the same changes as you did to `handleBlur` to this function. Won't we still have the timezone issue in forms with `mode="onChange"`?

Reply:
Updated